java/src/main/java/ai/rapids/cudf/MemoryBuffer.java

```diff
@@ -193,13 +209,35 @@ public final void copyFromMemoryBufferAsync(
    */
   public abstract MemoryBuffer slice(long offset, long len);
 
+  /**
+   * Set an event handler for this buffer. This method can be invoked with null
+   * to unset the handler.
+   */
+  public synchronized void setEventHandler(EventHandler handler) {
+    if (this.eventHandler != null && handler != null) {
+      throw new IllegalStateException("EventHandler is already set for this buffer");
+    }
```

Comment:
IMO this should not check or throw.  Instead this should return the old handler and let the caller decide if they want to assert or do other logic.